src/main/java/org/opensearch/neuralsearch/processor/NormalizationProcessorWorkflow.java

```diff
@@ -67,22 +78,78 @@ public void execute(
     private List<CompoundTopDocs> getQueryTopDocs(final List<QuerySearchResult> querySearchResults) {
         List<CompoundTopDocs> queryTopDocs = querySearchResults.stream()
             .filter(searchResult -> Objects.nonNull(searchResult.topDocs()))
-            .filter(searchResult -> searchResult.topDocs().topDocs instanceof CompoundTopDocs)
-            .map(searchResult -> (CompoundTopDocs) searchResult.topDocs().topDocs)
+            .map(querySearchResult -> querySearchResult.topDocs().topDocs)
+            .map(CompoundTopDocs::new)
             .collect(Collectors.toList());
+        if (queryTopDocs.size() != querySearchResults.size()) {
+            log.warn("Some of querySearchResults are not produced by hybrid query");
```

Comment:
if we see this warning what should the action? like in which cases this can happen? can we add some details in the warning about this.

Reply:
I think it's old case when results from different nodes were not serialized properly. I haven't seen this warning since we switched to a single list approach. We can probably fail for such case, as otherwise scores will be silently corrupted. Let me add exception and push new commit